lab6

AminaMahauri opened the pull request.

AminaMahauri committed:
Create лаба 6

AminaMahauri committed:
Update лаба 6

NemirovichMark commented:
Здравствуйте, Амина.

1й номер
Свойства нужны для предоставления информации другим программам, но их менять следует только внутри структуры, поэтому их нужно делать либо только для чтения, либо с приватным сеттером. Поправьте свойства в вашей программе.
public string Surname { get; }
public string Surname { get; private set; }

Вывод информации поместить внутрь структуры.
            Console.WriteLine($"{runner.Surname}\t\t{runner.Group}\t\t{runner.TeacherSurname}\t\t{runner.Result:F2}\t\t{(runner.Result <= 10 ? "Выполнен" : "Не выполнен")}");
2й номер
Публичные поля - нарушение принципов ООП ещё большее, чем свойства с публичным сеттером.
Сделайте их приватными, и для заполнения используйте конструктор:
 new Player { Name = "Player1", TotalPenaltyTime = 15 },
меняется на  new Player ("Player1", 15),

Вывод информации поместить внутрь структуры.
            Console.WriteLine($"{candidate.Name} - {candidate.TotalPenaltyTime} мин");
            
3й номер аналогичные замечания.

На доработку.

AminaMahauri committed:
исправленная lab6

NemirovichMark commented:
Здравствуйте, Амина.

Во всех номерах вывод информации поместить внутрь структур.
Console.WriteLine($"{candidate.Name} - {candidate.TotalPenaltyTime} мин");

Суть ООП - объект содержит в себе все данные, и рассказывает о них так, как считает нужным, а не мы извне достаем информацию по кусочкам.

На доработку.

AminaMahauri committed:
Create исправление №2 лаб 6

Здравствуйте, Марк Анатольевич!
Я исправила вывод ответа в первом номере, во 2 и 3 было в таком же формате, ка вы написали

NemirovichMark commented:
Хорошо!
Принял.

AminaMahauri commented:
УРААА! СПАСИБО
